### DISCUSSION TITLE
web-ui 컴포넌트 1123번 이슈 관련

### DISCUSSION THREAD

**Discussion by yujung7768903** [lablup/contribution-academy-2021#107] [Q&A] [answered]
관련 이슈 : https://github.com/lablup/backend.ai-webui/issues/1123

## 첫번째 질문
> 참고 링크 : https://github.com/lablup/backend.ai-webui/blob/main/src/components/backend-ai-dialog.ts#L165-L172
https://github.com/lablup/backend.ai-webui/blob/main/src/components/backend-ai-dialog.ts#L165-L172

약관또는 개인정보보호방침 dialog의 x버튼을 누르게 되면 backend-ai-dialog.ts 파일 안에 있는 _hideDialog()가 실행이 된 후에 hide() 함수가 실행됩니다. 
hide() 함수가 실행된 후에는 backend-ai-dialog의 부모인 lablup-terms-of-service 엘리먼트에 `aria-hidden=true`가 붙으면서 dialog가 보이지 않게 됩니다. 
**어떻게 부모 요소인 lablup-terms-of-service의 aria-hidden속성이 바뀌게 된 것인지, 이 과정이 잘 이해가 가지 않습니다.** 이 부분을 이해하면 이슈를 해결하는데 도움이 될 것 같습니다.
```
  hide() {
    if (this.closeWithConfirmation) {
      const closeEvent = new CustomEvent('dialog-closing-confirm', {detail: ''});
      this.dispatchEvent(closeEvent);
    } else {
      this.dialog.close();
    }
  }
```
#### 접근 방법
hide() 함수를 이해해야 `부모 태그가 보이지 않게 된 과정`을 이해할 수 있을 것 같아서 아래와 같은 방법으로 이해하려 노력했습니다.
* 먼저 if안에 있는 코드를 살펴봤습니다.
x버튼을 누르게 되면 'dialog-closing-confirm' 이벤트가 만들어지고, 부모인 lablup-terms-of-service 엘리먼트가 이 이벤트를 처리하면서 그 과정에서 aria-hidden이 true로 바뀌고 보이지 않게 되는 것이라 생각했습니다.
하지만 lablup-terms-of-serviece.ts 파일에서 관련된 이벤트 리스너를 찾지 못했습니다.
* 그래서 else안에 있는 `this.dialog.close()`함수를 살펴보려 했지만, close()함수가 정의되어 있는 부분을 찾지 못했습니다.
backend-ai-dialog.ts 에서 dialog는 아래처럼 정의되어 있습니다. 
```
@query('#dialog') protected dialog;
```
lit-html에서 @query는 this.renderRoot.querySelector('')와 같은 표현으로 shadow root안에 있는 node에 접근하기 위한 방법으로 알고 있습니다.
backend-ai-dialog.ts 파일에서 dialog는 아이디가 dialog인 mwc-dialog 엘리먼트를 의미한다고 파악했습니다.
그래서 this.dialog.close()의 close()함수는 mwc-dialog.ts 파일에 있을 것이라 생각했지만, 그렇지 않았습니다.

**Comment by lizable** [comment, marked as answer]
@yujung7768903  우선 `dialog-closing-confirm` 의 경우, 닫기 전 confirm modal을 보이게 하는 이벤트 리스너로 본 이슈와는 관련이 크게 없는 것으로 보입니다. confirm modal이 필요한 경우가 몇 가지 있는데, 특히 사용자 입력이 있는 modal(dialog) 같은 경우에는 close with confirmation이 필요하게 됩니다. (이 경우도 현재 일원화가 되어있지 않기 때문에 디자인 시스템과 함께 상황에 맞게 정립이 되어야 합니다.)

**Reply by lizable** [reply]
그리고, `aria-hidden` 속성의 경우는 custom 하게 바꾸는 부분은 따로 없습니다. 따라서, 다른 부분을 좀더 살펴보시는 것이 문제 해결에 도움이 될 수 있겠습니다.
